lib/Default/SmrPort.class.inc

```diff
@@ -523,15 +523,16 @@ class SmrPort {
 			throw new Exception('Cannot downgrade a cached port!');
 
 		switch($this->level) {
-			case 1: // we don't want to downgrade a level 1 port
-//				for($i=0;$i<3;$i++) { // Add 3 goods for upgrading to level 1
-//					do {
-//						$newGood = array_shift($GOODS);
-//					} while( (!$this->hasGood($newGood['ID']) || $newGood['Class']!=1) && ($count = count($GOODS))!=0 );
-//					if($count == 0)
-//						break;
-//					$this->removePortGood($newGood['ID'],);
-//				}
+			case 1:
+				// Replace one class 1 good with another
+				$this->selectAndRemoveGood(1);
+				$newGood = $this->selectAndAddGood(1);
+				// Set new good to 0 supply
+				// (since other goods are set to 0 when port is destroyed)
+				$this->setGoodAmount($newGood['ID'], 0);
+				// We don't want to make the port level 0, so stop here
+				$this->setUpgrade(0);
+				return;
 			break;
```

Comment:
You could drop this `break;` since the `return;` means you can never reach it

Reply:
Good point. I slightly changed the logic so that this case doesn't have an alternate return, and instead I wrap the de-leveling call in a `$this->level > 1` conditional.